src/main/java/com/ibm/disni/util/MemBuf.java

```diff
@@ -23,24 +23,13 @@
 
 import java.nio.ByteBuffer;
 
-import org.slf4j.Logger;
-
 public class MemBuf {
-	private static final Logger logger = DiSNILogger.getLogger();
-
-	private MemoryAllocation.MemType type;
 	private long address;
 	private ByteBuffer buffer;
-	private MemoryAllocation memAlloc;
-	private String classname;
 
-	public MemBuf(MemoryAllocation.MemType type, ByteBuffer buffer,
-			long address, MemoryAllocation memAlloc, String classname) {
-		this.type = type;
+	public MemBuf(ByteBuffer buffer) {
```

Comment:
Do we need this to be public? I assume only the MemoryAllocation class should instantiate MemBufs?